lib/src/download_utils.dart

```diff
@@ -114,28 +114,37 @@ class UrlChecker {
   UrlChecker() {
     addInternalHosts([
       'dart.dev',
-      RegExp(r'.*\.dart\.dev'),
       'flutter.dev',
-      RegExp(r'.*\.flutter\.dev'),
+      'flutter.io',
       'pub.dev',
-      RegExp(r'.*\.pub\.dev'),
       'dartlang.org',
-      RegExp(r'.*\.dartlang\.org'),
       'example.com',
-      RegExp(r'.*\.example.com'),
       'localhost',
     ]);
   }
 
   /// Specify hosts internal to Dart. Non-internal packages
   /// should not reference internal hosts.
+  ///
+  /// When a host is [String], we will expand the rule to include:
+  /// - exact hostname match (`^$host$`), and
+  /// - subdomain hostname match (`^.+\.$host$`).
   void addInternalHosts(Iterable<Pattern> hosts) {
-    _internalHosts.addAll(hosts);
+    _internalHosts.addAll(hosts.expand((host) {
+      if (host is String) {
+        return [
+          RegExp('^$host\$'),
```

Comment:
I think you need to use `https://api.dart.dev/stable/2.17.5/dart-core/RegExp/escape.html` to escape the string before inserting it in a RegExp - otherwise 'pub.dev' would also match 'pubAdev

Also in the line below.

Reply:
Good catch, fixed it + added test.